test/spec/modules/userId_spec.js

```diff
@@ -2653,5 +2653,97 @@ describe('User ID', function () {
         });
       });
     });
+
+    describe('handles config with ESP configuration in user sync object', function() {
+      describe('Call registerSignalSources to register signal sources with gtag', function () {
+        it('pbjs.registerSignalSources should be defined', () => {
+          expect(typeof (getGlobal()).registerSignalSources).to.equal('function');
+        });
+      })
+
+      describe('Call getEncryptedEidsForSource to get encrypted Eids for source', function() {
+        const signalSources = ['pubcid.org'];
+
+        it('pbjs.getEncryptedEidsForSource should be defined', () => {
+          expect(typeof (getGlobal()).getEncryptedEidsForSource).to.equal('function');
+        });
+
+        it('pbjs.getEncryptedEidsForSource should return the string without encryption if encryption is false', (done) => {
+          setSubmoduleRegistry([sharedIdSystemSubmodule]);
+          init(config);
+          config.setConfig({
+            userSync: {
+              syncDelay: 0,
+              userIds: [
+                {
+                  'name': 'sharedId',
+                  'storage': {
+                    'type': 'cookie',
+                    'name': '_pubcid',
+                    'expires': 365
+                  }
+                },
+                {
+                  'name': 'pubcid.org'
+                }
+              ]
+            },
+          });
+          const encrypt = false;
+          (getGlobal()).getEncryptedEidsForSource(signalSources[0], encrypt).then((data) => {
+            let users = (getGlobal()).getUserIdsAsEids();
+            expect(data).to.equal(users[0].uids[0].id);
+            done();
+          }).catch(done);
+        });
+
+        it('pbjs.getEncryptedEidsForSource should return the string base64 encryption if encryption is true', (done) => {
+          const encrypt = true;
+          (getGlobal()).getEncryptedEidsForSource(signalSources[0], encrypt).then((result) => {
+            expect(result.startsWith('1||')).to.true;
+            done();
+          }).catch(done);
+        });
+
+        it('pbjs.getEncryptedEidsForSource should return string if custom function is defined', () => {
+          const getCustomSignal = () => {
+            return '{"keywords":["tech","auto"]}';
+          }
+          const expectedString = '"1||{\"keywords\":[\"tech\",\"auto\"]}"';
+          const encrypt = false;
+          const source = 'pubmatic.com';
+          (getGlobal()).getEncryptedEidsForSource(source, encrypt, getCustomSignal).then((result) => {
+            expect(result).to.equal(expectedString);
+            done();
```

Comment:
@nitin0610 - this test does not pass if you let it run until the assertion (`done` is undefined here).  Can you fix it so I understand how I need to adjust https://github.com/prebid/Prebid.js/pull/8201 ? Thanks!

Reply:
@dgirardi Have fixed the issue. Raised PR https://github.com/prebid/Prebid.js/pull/8212 for the same